tls/s2n_prf.c

```diff
@@ -459,43 +461,129 @@ S2N_RESULT s2n_prf_free(struct s2n_connection *conn)
     return S2N_RESULT_OK;
 }
 
-static int s2n_prf(struct s2n_connection *conn, struct s2n_blob *secret, struct s2n_blob *label, struct s2n_blob *seed_a,
-        struct s2n_blob *seed_b, struct s2n_blob *seed_c, struct s2n_blob *out)
+static bool s2n_libcrypto_supports_tls_prf()
 {
-    POSIX_ENSURE_REF(conn);
-    POSIX_ENSURE_REF(secret);
-    POSIX_ENSURE_REF(conn->prf_space);
-    POSIX_ENSURE_REF(conn->secure);
-
-    /* seed_a is always required, seed_b is optional, if seed_c is provided seed_b must also be provided */
-    S2N_ERROR_IF(seed_a == NULL, S2N_ERR_PRF_INVALID_SEED);
-    S2N_ERROR_IF(seed_b == NULL && seed_c != NULL, S2N_ERR_PRF_INVALID_SEED);
-
-    if (conn->actual_protocol_version == S2N_SSLv3) {
-        return s2n_sslv3_prf(conn, secret, seed_a, seed_b, seed_c, out);
-    }
+    return S2N_LIBCRYPTO_SUPPORTS_TLS_PRF;
+}
 
+static S2N_RESULT s2n_custom_prf(struct s2n_connection *conn, struct s2n_blob *secret, struct s2n_blob *label, struct s2n_blob *seed_a,
```

Comment:
Nit: This line is 135 characters long, so probably needs to be wrapped earlier. I'd review the line wrap for the other code you touched too; some of the other lines are also really long. Github is wrapping them on my screen and making the code harder to read 😅

Reply:
Sorry - fixed! I wrapped everything to 120, is that enough? I wonder if clang-format should check this..

Reply:
We intentionally didn't enable that in clang-format bc it is TERRIBLE at automated line wrapping xD